File: simplified-books-core-ng/src/main/java/org/nypl/simplified/books/accounts/AccountBundledCredentialsType.java
package org.nypl.simplified.books.accounts;

import com.io7m.jfunctional.OptionType;

import java.net.URI;
import java.util.Map;

/**
 * A set of credentials bundled with the application.
 */

public interface AccountBundledCredentialsType {

  /**
   * @return A read-only map of the bundled credentials
   */

  Map<URI, AccountAuthenticationCredentials> bundledCredentials();

  /**
   * Obtain bundled credentials when creating account using the given provider.
   *
   * @param accountProvider The account provider URI
   * @return The bundled credentials, if any
   */

  OptionType<AccountAuthenticationCredentials> bundledCredentialsFor(
    URI accountProvider);

}

File: simplified-accounts-api/src/main/java/org/nypl/simplified/accounts/api/AccountBundledCredentialsType.kt
package org.nypl.simplified.accounts.api

import java.net.URI

/**
 * A set of credentials bundled with the application.
 */

interface AccountBundledCredentialsType {

  /**
   * @return A read-only map of the bundled credentials
   */

  fun bundledCredentials(): Map<URI, AccountAuthenticationCredentials>

  /**
   * Obtain bundled credentials when creating account using the given provider.
   *
   * @param accountProvider The account provider URI
   * @return The bundled credentials, if any
   */

  fun bundledCredentialsFor(
    accountProvider: URI
  ): AccountAuthenticationCredentials?
}
